src/main/java/randoop/generation/GrtObjectFuzzer.java

```diff
@@ -0,0 +1,382 @@
+   *
+   * <ol>
+   *   <li>At least one parameter position matches the target variable's type
+   *   <li>The target variable's sequence is incorporated as input for the mutation
+   * </ol>
+   *
+   * Without this selection, we might generate mutations that don't exercise the target variable.
+   *
+   * @param formals the formal parameter types of the mutation operation
+   * @param typeToFuzz the type of the target variable to fuzz
+   * @param mutationOp the operation being mutated (used for error reporting)
+   * @return the index of the selected parameter position
+   * @throws RandoopBug if no compatible parameter positions exist (indicating improper operation
+   *     filtering)
+   */
+  private int selectFuzzParameterPosition(
+      TypeTuple formals, Type typeToFuzz, TypedOperation mutationOp) {
+    List<Integer> candidateParamPositions = new ArrayList<>();
+    for (int i = 0; i < formals.size(); i++) {
+      if (formals.get(i).isAssignableFrom(typeToFuzz)) {
+        candidateParamPositions.add(i);
+      }
+    }
+
+    // No candidate positions found for the type to fuzz.
+    // This should not happen, as we have already checked that the operation is applicable to the
+    // type to fuzz.
+    if (candidateParamPositions.isEmpty()) {
+      throw new RandoopBug(
+          "No candidate positions found for the type "
+              + typeToFuzz
+              + " in the operation "
+              + mutationOp
+              + ". This should not happen.");
+    }
+    return Randomness.randomMember(candidateParamPositions);
+  }
+
+  /**
+   * Returns a canonical/erased view of a type. For parameterized class/interface types, this
+   * returns the raw type; otherwise it returns the type unchanged.
+   *
+   * @param t the type to erase (non-null)
+   * @return the erased/canonical type
+   */
+  private Type erase(Type t) {
+    if (t == null) {
+      throw new IllegalArgumentException("type cannot be null");
+    }
+    Type raw = t.getRawtype(); // Randoop's API: raw type for generics
+    return (raw != null) ? raw : t;
+  }
+
+  /**
+   * Performs a breadth-first traversal over the given type and its ancestors (superclasses and
+   * interfaces), returning the erased form of each in order.
+   *
+   * <p>Order: the starting type first, then superclass, then interfaces, then their ancestors.
+   * Duplicates are removed while preserving order.
+   *
+   * @param t the starting type
+   * @return a list of erased types including {@code t} and all ancestors
+   */
+  private List<Type> ancestorsAndSelf(Type t) {
+    Type start = erase(t);
+
+    // Non-class/interface types (primitives, arrays) have no ancestors.
+    if (!(start instanceof ClassOrInterfaceType)) {
+      List<Type> singleton = new ArrayList<>(1);
+      singleton.add(start);
+      return singleton;
+    }
+
+    java.util.Deque<Type> queue = new java.util.ArrayDeque<>();
```

Comment:
Given that this is used as a queue, a better declaration is
```suggestion
    Queue<Type> queue = new java.util.ArrayDeque<>();
```
Don't forget to import `java.util.Queue`.